### DISCUSSION TITLE
Feature request: Inbound shipments additional data

### DISCUSSION THREAD

**Discussion by phstc** [amzn/selling-partner-api-models#4289] [Ideas]
I would like to request additional data for inbound shipments.

- Inbound shipments
    - Units located
    - Units reimbursed
    - Associated support case (if any)
    - Shipment Events (with respective dates):
        - Shipment created
        - In transit
        - Delivered
        - Checked in
        - Received
        - Closed

Having access to these data points is crucial for determining the state of each shipment and if there are any missing units after reconciliation. 

Thank you!

**Comment by puppsupr** [comment]
Hi @phstc,
Have you looked at our [FBA Reports](https://developer-docs.amazon.com/sp-api/docs/report-type-values-fba). This should give you most of the requested data.

**Reply by phstc** [reply]
Hi @puppsupr 

Thank you for your quick reply! 

I'm struggling to find that information on the FBA Reports. 

Supposed that the shipment FBA123 has quantity expected 10 for SKU123, and 5 was received, then 3 is located, then 2 is missing. How can I know that using those reports?

**Comment by puppsupr** [comment]
Hi @phstc,
You can use the [FBA Inbound Performance Report](https://developer-docs.amazon.com/sp-api/docs/report-type-values-fba#fba-inbound-performance-report) and the [FBA Stranded Inventory Report](https://developer-docs.amazon.com/sp-api/docs/report-type-values-fba#fba-stranded-inventory-report).

**Reply by phstc** [reply]
Thank you @puppsupr 

That's very helpful. The performance one is almost there. Unfortunately, it does not include counted and confirmed, or when a case is already submitted, it does not list the discrepancy, no matter if the discrepancy is still not fixed. 

Also, is there a way to retrieve the respective case ID?

**Reply by phstc** [reply]
The case ID I'm talking about:

![image](https://github.com/user-attachments/assets/20720dc4-d201-45de-8fa0-accbdbd57523)

**Reply by phstc** [reply]
@puppsupr about the performance report, it doesn't seem to be up to date. Seller Central flags 2 missing, and the report shows 21.

<img width="543" height="345" alt="image" src="https://github.com/user-attachments/assets/1c27abc1-6da4-48df-87db-d6e3e65ca57e" />

<img width="1542" height="215" alt="image" src="https://github.com/user-attachments/assets/b8c7980f-3ba4-4ec5-8b23-45fc80b582a2" />

**Comment by cohlar** [comment]
@phstc have you found a way to get FBA shipment's event dates via the API?

**Reply by phstc** [reply]
@cohlar I have not. I wish the SP-API could expose all the info we have on Seller Central.

**Reply by phstc** [reply]
@BrunnoPigatto also looping you here in case you have any updates on the availability of shipment data through the API.

It would be incredibly helpful for our sellers if the shipment API (or any other API) could expose the same shipment details available on Seller Central, but through the API.

This would ensure sellers and providers can keep track of their shipment statuses. This currently takes a significant amount of laborious manual work. 

From the original request https://github.com/amzn/selling-partner-api-models/discussions/4289#discussion-7420277

```
Inbound shipments

    Units located
    Units reimbursed
    Associated support case (if any)
    Shipment Events (with respective dates):
        Shipment created
        In transit
        Delivered
        Checked in
        Received
        Closed

```
